src/lib/helpers/ArrowHelper.svelte

```diff
@@ -0,0 +1,52 @@
+<script lang="ts">
+  import type { ArrowHelperProperties, Object3DProperties } from '$lib/types/components';
+  import { ArrowHelper as ThreeArrowHelper } from 'three'
+  import Object3DInstance from '../instances/Object3DInstance.svelte'
+
+  // Object3DInstance
+  export let position: Object3DProperties['position'] = undefined
+  export let scale: Object3DProperties['scale'] = undefined
+  export let rotation: Object3DProperties['rotation'] = undefined
+  export let lookAt: Object3DProperties['lookAt'] = undefined
+  export let viewportAware: Object3DProperties['viewportAware'] = false
+  export let inViewport: Object3DProperties['inViewport'] = false
+  export let castShadow: Object3DProperties['castShadow'] = undefined
+  export let receiveShadow: Object3DProperties['receiveShadow'] = undefined
+  export let frustumCulled: Object3DProperties['frustumCulled'] = undefined
+  export let renderOrder: Object3DProperties['renderOrder'] = undefined
+  export let visible: Object3DProperties['visible'] = undefined
+  
+  //self
+  export let dir:ArrowHelperProperties['dir'] = undefined;
+  export let origin:ArrowHelperProperties['origin'] = undefined;
+  export let length:ArrowHelperProperties['length'] = undefined;
+  export let color:ArrowHelperProperties['color'] = undefined;
+  export let headLength:ArrowHelperProperties['headLength'] = undefined;
+  export let headWidth:ArrowHelperProperties['headWidth'] = undefined;
+
+  let helper: ThreeArrowHelper = new ThreeArrowHelper (dir,origin,length,color,headLength,headWidth);
+
+  $:{
+    if(length!==undefined || headLength!==undefined || headWidth!==undefined) helper.setLength(length,headLength,headWidth);
```

Comment:
We might want to warn the user if *some* of these properties were supplied but not all.

Reply:
Also a very finicky detail but while we're at it:
This reactive block is rerunning on any component prop change that is referenced. If the color is changing, we might not want to call `setLength` or `setDir` as that could be an expensive operation, so make that a separate reactive statement for each function call:

```ts
$: setLength…
$: setColor…
$: setDir…
```

Reply:
> We might want to warn the user if _some_ of these properties were supplied but not all.

Would that be done through the `invalidate` function? I've noticed other components use it to send debug messages to the user. Is that what you mean?

Reply:
Ah true, i totally missed that part! Every time a prop changes and the component takes that into account e.g. by calling methods like `setLength` we need to invalidate the current frame by calling `invalidate(debugMessage?: string)` you should pass a debugmessage as that makes it easier to spot performance bottlenecks when using `<Canvas debugFrameloop>`.